modules/core/include/visp3/core/vpMeterPixelConversion.h

```diff
@@ -240,6 +246,94 @@ class VISP_EXPORT vpMeterPixelConversion
     iP.set_u(cam.u0 + cam.px * x * r2);
     iP.set_v(cam.v0 + cam.py * y * r2);
   }
+
+  /*!
+    Point coordinates conversion with Kannala-Brandt distortion from
+    normalized coordinates \f$(x,y)\f$ in meter to pixel coordinates
+    \f$(u,v)\f$.
+
+    \param[in] cam : camera parameters.
+    \param[in] x   : input coordinate in meter along image plane x-axis.
+    \param[in] y   : input coordinate in meter along image plane y-axis.
+    \param[out] u  : output coordinate in pixels along image horizontal axis.
+    \param[out] v  : output coordinate in pixels along image vertical axis.
+
+    \f$ r = sqrt{x^2 + y^2} \f$
+    \f$ \theta = \arctan{r} \f$
+    Calculate \f$ r_d \f$ knowing ditorsion coefficients as follows:
+    \f$ r_d = \theta + k_1 \theta^3 + k_2 \theta^5 + k_3 \theta^7 + k_4 \theta^9 \f$
+    \f$ scale = r_d / r \f$
+    \f$ x_d = x * scale \f$
+    \f$ y_d = y * scale \f$
+    \f$ u = x_d*p_x+u_0 \f$ and
+    \f$ v = y_d*p_y+v_0 \f$
+    with \f$ r^2 = x^2+y^2 \f$
+  */
+  inline static void convertPointWithKannalaBrandtDistortion(const vpCameraParameters &cam, const double &x, const double &y,
+                                                             double &u, double &v)
+  {
+    double r = sqrt(vpMath::sqr(x) + vpMath::sqr(y));
+    double theta = atan(r);
+
+    std::vector<double> k = cam.getKannalaBrandtDistortionCoefficients();
+
+    double theta2 = theta*theta, theta3 = theta2*theta, theta4 = theta2*theta2, theta5 = theta4*theta,
+          theta6 = theta3*theta3, theta7 = theta6*theta, theta8 = theta4*theta4, theta9 = theta8*theta;
+
+    double r_d = theta + k[0]*theta3 + k[1]*theta5 + k[2]*theta7 + k[3]*theta9;
+
+    double scale = (r == 0) ? 1.0 : r_d / r;
+
```

Comment:
Maybe here we will get a warning during build like comparing to zero is not good.
To avoid this future warning:
```
 double scale = (std::fabs(r) < std::numeric_limits<double>::epsilon()) ? 1.0 : r_d / r;
```